### PR TITLE
Audit3. Refactor council budget handling.

### PR DESCRIPTION
**Pull request by shamil-gadelshin**
We need to refactor the council budget handling to reduce code duplication.

#### Changes
- added new methods for the `CouncilBudgetManager` trait:
  - try_transfer
  - transfer
  - increase_budget
- changed the `bounty` palelt to use new methods

[Original issue](https://github.com/Joystream/joystream/issues/3496)

### CODE REVIEW

File: runtime-modules/common/src/council.rs [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
     fn get_budget() -> Balance;
 
     /// Set the current budget value.
     fn set_budget(budget: Balance);
+
+    /// Remove some balance from the council budget and transfer it to the account. Fallible.
+    fn try_transfer(account_id: &AccountId, amount: Balance) -> DispatchResult;
+
+    /// Remove some balance from the council budget and transfer it to the account. Infallible.
+    fn transfer(account_id: &AccountId, amount: Balance) {
```
Comment (bedeho) [inline runtime-modules/common/src/council.rs]: `withdraw` is a bit more explicit

Comment (bedeho) [reply runtime-modules/common/src/council.rs]: Can you also explicitly say that if the budget does not cover full withdrawal, there is no side-effect.

**Review by bedeho — COMMENTED** [review]
Nice work, just a cosmetic tweak.

File: runtime-modules/common/src/council.rs (hunk 1 above)
Comment (shamil-gadelshin) [reply runtime-modules/common/src/council.rs]: Addressed both comments: https://github.com/Joystream/joystream/pull/3516/commits/ad221feb144e467f4422acd273b058fbc6692ff1

**Review by bedeho — APPROVED** [review]
LGTM.

**Comment by shamil-gadelshin** [comment]
Superceded by https://github.com/Joystream/joystream/pull/3518

**Pull request closed without merging**